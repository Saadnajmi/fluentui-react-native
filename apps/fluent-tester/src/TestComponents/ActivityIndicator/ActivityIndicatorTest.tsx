import * as React from 'react';
import { View, Switch, PlatformColor } from 'react-native';

import { Text } from '@fluentui/react-native';
import { ActivityIndicator } from '@fluentui-react-native/experimental-activity-indicator';
import { Stack } from '@fluentui-react-native/stack';
import { VibrancyView } from '@fluentui-react-native/vibrancy-view';

import { ACTIVITY_INDICATOR_TESTPAGE } from '../../../../E2E/src/ActivityIndicator/consts';
import { stackStyle, commonTestStyles as commonStyles } from '../Common/styles';
import type { TestSection, PlatformStatus } from '../Test';
import { Test } from '../Test';
import { width } from '@fortawesome/free-solid-svg-icons/faMountainCity';

const BasicActivityIndicator: React.FunctionComponent = () => {
  const [animating, setAnimating] = React.useState(true);
  const [hidesWhenStopped, setHidesWhenStopped] = React.useState(true);

  return (
    <Stack style={stackStyle}>
      <View style={[commonStyles.root, { width: 80, height: 500, backgroundColor: 'black' }]}>
        <VibrancyView
          material={'menu'}
          blendingMode={'behindWindow'}
          style={{ width: 200, height: 400, borderColor: 'red', borderWidth: 1, borderRadius: 8 }}
        >
          {/* <VisualEffectView
            allowsVibrancy={true}
            behindWindow={false}
            material={'popover'}
            style={{ width: 200, height: 150, borderColor: 'red', borderWidth: 1, borderRadius: 8 }}
          > */}
          <View allowsVibrancy={true} style={[commonStyles.root, { width: 150, height: 100, backgroundColor: 'blue' }]}>
            <View allowsVibrancy={false} style={[commonStyles.root, { width: 200, height: 50, backgroundColor: 'red' }]} />
          </View>
        </VibrancyView>
        {/* </VisualEffectView> */}
      </View>
    </Stack>
  );
};

const ActivityIndicatorMainTest: React.FunctionComponent = () => {
  return (
    <Stack style={stackStyle}>
      <Text>Extra Small</Text>
      <ActivityIndicator size="xSmall" />
      <Text>Small</Text>
      <ActivityIndicator size="small" />
      <Text>Medium</Text>
      <ActivityIndicator size="medium" />
      <Text>Large</Text>
      <ActivityIndicator size="large" />
      <Text>Extra Large</Text>
      <ActivityIndicator size="xLarge" />

      <Text>Size=xLarge and Line Thickness=large</Text>
      <ActivityIndicator size="xLarge" lineThickness="large" />
      <Text>Color props</Text>
      <ActivityIndicator activityIndicatorColor="orange" accessibilityLabel="orange progressbar" />
    </Stack>
  );
};

const CustomizedActivityIndicator = ActivityIndicator.customize({
  activityIndicatorColor: 'orange',
  size: 'large',
});

const CustomizedActivityIndicatorTest: React.FunctionComponent = () => {
  return (
    <Stack style={stackStyle}>
      <Text>Customized Activity Indicator</Text>
      <CustomizedActivityIndicator />
    </Stack>
  );
};

const activityIndicatorSections: TestSection[] = [
  {
    name: 'Base ActivityIndicator',
    testID: ACTIVITY_INDICATOR_TESTPAGE,
    component: BasicActivityIndicator,
  },
  {
    name: 'ActivityIndicator',
    testID: ACTIVITY_INDICATOR_TESTPAGE,
    component: ActivityIndicatorMainTest,
  },
  {
    name: 'Customized ActivityIndicator',
    testID: ACTIVITY_INDICATOR_TESTPAGE,
    component: CustomizedActivityIndicatorTest,
  },
];

export const ActivityIndicatorTest: React.FunctionComponent = () => {
  const status: PlatformStatus = {
    win32Status: 'Backlog',
    uwpStatus: 'Backlog',
    iosStatus: 'Beta',
    macosStatus: 'Backlog',
    androidStatus: 'Beta',
  };

  const description =
    'ActivityIndicator is a visual representation that data is being loaded. It is implemented with a View wrapping an Animated SVG. The View is to ensure that AccessibilityRole works. AccessibilityRole currently does not work on SVGs.';

  return <Test name="ActivityIndicator Test" description={description} sections={activityIndicatorSections} status={status}></Test>;
};
