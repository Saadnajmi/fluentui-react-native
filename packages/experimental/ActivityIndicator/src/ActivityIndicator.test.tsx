import * as React from 'react';
import { ActivityIndicator } from './ActivityIndicator';
import * as renderer from 'react-test-renderer';

it('ActivityIndicator default', () => {
  const tree = renderer.create(<ActivityIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});
