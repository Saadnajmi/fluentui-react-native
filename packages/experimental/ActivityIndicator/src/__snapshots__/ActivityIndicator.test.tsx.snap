// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator default 1`] = `
<View
  accessibilityLabel="progressbar"
  accessibilityState={
    Object {
      "busy": true,
    }
  }
  accessible={true}
  activityIndicatorColor="#BDBDBD"
  lineThickness="medium"
  size="medium"
  style={
    Object {
      "height": 24,
      "opacity": 1,
      "width": 24,
    }
  }
>
  <RNSVGSvgView
    bbHeight={24}
    bbWidth={24}
    focusable={false}
    height={24}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "transform": Array [
            Object {
              "rotateZ": "0deg",
            },
            Object {
              "perspective": 10,
            },
          ],
        },
        Object {
          "flex": 0,
          "height": 24,
          "width": 24,
        },
      ]
    }
    width={24}
  >
    <RNSVGGroup
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
    >
      <RNSVGPath
        d="M1 12 a11 11 0 1 1 11 11"
        propList={
          Array [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        stroke={-4342339}
        strokeLinecap={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
